tools/devops/device-tests.yml

```diff
@@ -0,0 +1,236 @@
+resources:
+  repositories:
+  - repository: maccore
+    type: github
+    name: xamarin/maccore
+    ref: refs/heads/device-tests-yaml
+    endpoint: xamarin
+
+jobs:
+- job: macOS
+  displayName: xamarin macios
+  timeoutInMinutes: 1000
+
+  pool:
+    name: VSEng-Xamarin-Mac-Devices
+    demands: sh
+
+  steps:
+  - checkout: self
+  - checkout: maccore
+
+  - bash: ./xamarin-macios/tools/devops/system-info.sh
+    displayName: System info
+
+  - bash: cd xamarin-macios && ./configure --enable-xamarin
+    displayName: 'Enable Xamarin'
+
+  - script: |
+     set -x
+     set -e
+
+     rm -f ~/Library/Caches/com.xamarin.provisionator/Provisions/*p12
+     rm -f ~/Library/Caches/com.xamarin.provisionator/Provisions/*mobileprovision
+
+     ./maccore/tools/install-qa-provisioning-profiles.sh -v
+    displayName: 'Add provisioning profiles'
+    env:
+      LOGIN_KEYCHAIN_PASSWORD: $(OSX_KEYCHAIN_PASS)
+
+  - script: |
+     set -x
+     set -e
+
+     CONFIGURE_ARGS="--provision-xcode --provision-xamarin-studio --provision-mono --ignore-osx --ignore-autotools --ignore-cmake"
+     if grep ignore-7z xamarin-macios/system-dependencies.sh 2>&1 > /dev/null; then
+         CONFIGURE_ARGS="$CONFIGURE_ARGS --provision-7z"
+     fi
+     if grep ignore-python3 xamarin-macios/system-dependencies.sh 2>&1 > /dev/null; then
+         CONFIGURE_ARGS="$CONFIGURE_ARGS --ignore-python3"
+     fi
+     ./xamarin-macios/system-dependencies.sh $CONFIGURE_ARGS
+    displayName: 'Provision dependencies'
+    timeoutInMinutes: 240
+
+  - task: ms-devlabs.utilitytasks.task-Shellpp.Shell++@0
+    displayName: 'Install XI PKG'
+    inputs:
+      type: InlineScript
+      script: |
+       #!/bin/bash -ex
```

Comment:
I think bigger scripts would be better as external .sh files, preferably in tools/devops/

Reply:
Tots agree, let me clean that up!

Reply:
yep, same opinion. I wish we could have a language that can be tested easily.